extension/json/src/include/reader/buffered_json_reader.h

```diff
@@ -0,0 +1,85 @@
+#include "common/file_system/file_info.h"
+#include "common/json_common.h"
+#include "common/json_enums.h"
+#include "main/client_context.h"
+
+namespace kuzu {
+namespace json_extension {
+
+struct JsonScanBuffer {
+    std::unique_ptr<uint8_t[]> data;
+
+    JsonScanBuffer() : data{std::make_unique<uint8_t[]>(JsonConstant::SCAN_BUFFER_CAPACITY)} {}
+
+    uint8_t* getData() const { return data.get(); }
+
+    uint64_t getCapacity() { return JsonConstant::SCAN_BUFFER_CAPACITY; }
+};
+
+struct JsonScanBufferHandle {
+    uint64_t bufferIdx;
```

Comment:
I would suggest using `common::idx_t` as much as u can. Although it's `uint32_t` so double check if it can be used here.